src/Blocks/components/form/assets/form.js

```diff
@@ -62,20 +62,101 @@ export class Form {
+		} else {
+			// Find all forms elements
+			const forms = document.querySelectorAll(this.state.getStateSelectorsForms());
+			if (!forms) {
+				return;
+			}
+
+			[...forms].forEach((formsItems) => {
+				// Find all forms elements that have have geolocation data attribute.
+				if (formsItems?.getAttribute(this.state.getStateAttribute('formGeolocation'))) {
+					// If forms element have geolocation data attribute, init geolocation via ajax.
+					this.initGolocationForm(formsItems);
+				} else {
+					// If forms element don't have geolocation data attribute, init forms the regular way.
+					this.initOnlyFormsInner(formsItems?.children?.[0]?.getAttribute(this.state.getStateAttribute('formId')) || 0);
+				}
+			});
+		}
+	}
+
+	/**
+	 * Init only geolocation forms by ajax.
+	 * @param {object} formsElement Forms element.
+	 */
+	initGolocationForm(formsElement) {
+		const forms = formsElement?.querySelectorAll(this.state.getStateSelectorsForm());
+
+		const body = {
+			method: 'POST',
+			mode: 'same-origin',
+			headers: {
+				Accept: 'application/json',
+			},
+			body: formsElement?.getAttribute(this.state.getStateAttribute('formGeolocation')),
+			credentials: 'same-origin',
+			redirect: 'follow',
+			referrer: 'no-referrer',
+		};
 
-			setStateFormInitial(formId);
+		// Get geolocation data from ajax to detect what we will remove from DOM.
+		fetch(this.state.getRestUrl(ROUTES.GEOLOCATION), body)
+		.then((response) => {
+			this.formSubmitErrorContentType(response, 'geolocation');
+			return response.json();
+		})
+		.then((response) => {
+			// Get formId from ajax response.
+			let formId = response?.data?.formId;
+
+			// If form id is not set, get it from first form element.
+			if (!formId) {
+				formId = forms?.[0]?.getAttribute(this.state.getStateAttribute('formId')) || 0;
+			}
 
-			this.initOne(formId);
+			// Loop all form elements and remove all except the one we need.
+			[...forms].forEach((form) => {
+				if (form.getAttribute(this.state.getStateAttribute('formId')) !== formId) {
+					// Remove all forms except the one we got from ajax.
+					form.remove();
+				} else {
+					// Init form id that we got from ajax.
+					this.initOnlyFormsInner(formId);
 
-			this.conditionalTags.initOne(formId);
+					// Remove geolocation data attribute from forms element.
+					formsElement.removeAttribute(this.state.getStateAttribute('formGeolocation'));
+				}
+			});
 
-			this.steps.initOne(formId);
+			// Remove loading class from forms element.
+			formsElement?.classList?.remove(this.state.getStateSelectorsClassGeolocationLoading());
 		});
 	}
 
+	/**
+	 * Init only forms - inner items.
+	 * 
+	 * @returns {void}
+	 */
+	initOnlyFormsInner(formId) {
+		// Set state initial data for form.
+		setStateFormInitial(formId);
+
+		// Init all form elements.
+		this.initOne(formId);
```

Comment:
```suggestion
		// Init all form elements.
		this.initOne(formId);
```
I don't have the full context but the comment may be wrong here 😄

Reply:
the comment is correct here